invenio_communities/communities/services/components.py

```diff
@@ -169,3 +172,78 @@ def create(self, identity, data=None, record=None, **kwargs):
 
         # Invalidate the membership cache
         on_membership_change(identity=identity)
+
+
+class OAISetComponent(ServiceComponent):
+    """Service component for OAI set integration."""
+
+    _oai_sets_prefix = "community"
+
+    def _retrieve_set(self, record, **kwargs):
+        return OAISet.query.filter(OAISet.spec == self._create_set_spec(record.get("id"))).first()
+
+    def _create_set_spec(self, community_id):
+        return "{prefix}-{id}".format(prefix=self._oai_sets_prefix, id=community_id)
+
+    def _create_set_description(self, community_title):
+        return "Records belonging to the community '{title}'".format(title=community_title)
+
+    def _create_set_from_community(self, record, **kwargs):
+        community_id = kwargs.get("id") or record.get("id")
```

Comment:
I think this function can be reduced to only accepting only the record and do ``record.pid.pid_value``